packages/ember-runtime/lib/system/core_object.js

```diff
@@ -540,7 +541,9 @@ CoreObject.PrototypeMixin = Mixin.create({
   toString() {
     let hasToStringExtension = typeof this.toStringExtension === 'function';
     let extension = hasToStringExtension ? `:${this.toStringExtension()}` : '';
-    let ret = `<${this[NAME_KEY] || this.constructor.toString()}:${guidFor(this)}${extension}>`;
+    let factoryManager = peekFactoryManager(Object.getPrototypeOf(this));
+    let factoryToString = this[NAME_KEY] || (factoryManager && factoryManager.factoryToString());
```

Comment:
if `factoryToString()` was just toString() you wouldn't need to call it you can just let the string template call it.